### PR TITLE
Example refactor

### PR DESCRIPTION
**Pull request by rambleraptor**
This is a refactor of Ansible examples. This should help make this more maintainable for the future and help greatly in building facts examples from regular examples.

Highlights:

* Templates are now ERB instead of Ruby strings
* Properties are passed in as a hash instead of a giant string block.
* Properties are now put through a YAML parser
* Reorganized example.rb file + put better comments.
* Renamed the example template to integration_test because that's what the particular template does.

-----------------------------------------------------------------
# [all]

### CODE REVIEW

**Comment by rambleraptor** [comment]
This was probably because I rebased the mm repo off of regular Ansible. Let
me rerun the Magician and this should look a lot saner.

**Comment by rambleraptor** [comment]
Exciting things are happening! This PR now includes the ability to automatically create `facts` examples + tests from a regular Ansible example.

(This was the whole idea behind doing a refactor on examples)

With this, building facts modules is now completely free! No additional work has to be done to make a fact module when building a regular module.

File: provider/ansible/example.rb [hunk 1]
```diff
@@ -33,15 +33,86 @@ module Ansible
...
+        build_task(state, INTEGRATION_TEST_DEFAULTS, object, noop)
+      end
+
+      def build_example(state, object)
+        build_task(state, EXAMPLE_DEFAULTS, object)
+      end
+
+      private
+
+      # rubocop:disable Lint/UnusedMethodArgument
```
Comment (danawillow) [inline provider/ansible/example.rb]: Add a comment as to why this is disabled?

File: provider/ansible/example.rb [hunk 2]
```diff
@@ -54,41 +125,14 @@ def validate
         check_property :failure, FailureCondition
       end
 
-      # rubocop:disable Metrics/AbcSize
-      # rubocop:disable Metrics/MethodLength
+      # rubocop:disable Lint/UnusedMethodArgument
```
Comment (danawillow) [inline provider/ansible/example.rb]: here too

File: provider/ansible/example.rb (hunk 1 above)
Comment (rambleraptor) [reply provider/ansible/example.rb]: Yep, done!

TL;DR: The method arguments are used in the ERB files. We should figure out a better way to fix this. I know @ndmckinley has talked about it in the past.
